otx/api/utils/segmentation_utils.py

```diff
@@ -37,13 +39,26 @@ class index is zero. labels[0] matches pixel value 1, etc. The class index is
         Numpy array of mask
     """
     # todo: cache this so that it does not have to be redone for all the same media
-    mask = mask_from_annotation(
-        dataset_item.get_annotations(),
-        labels,
-        dataset_item.width,
-        dataset_item.height,
-    )
+    if use_otx_adapter:
+        mask = mask_from_annotation(dataset_item.get_annotations(), labels, dataset_item.width, dataset_item.height)
+    else:
+        mask = mask_from_file(dataset_item)
+    return mask
+
 
+def mask_from_file(dataset_item: DatasetItemEntity) -> np.ndarray:
+    """Loads masks directly from annotation image.
+
+    Only Common Sematic Segmentation format is supported.
+    """
+
+    mask_form_file = dataset_item.media.path
+    if mask_form_file is not None:
+        mask_form_file = mask_form_file.replace("images", "masks")
+        mask = cv2.imread(mask_form_file, cv2.IMREAD_GRAYSCALE)
+        mask = np.expand_dims(mask, axis=2)
```

Comment:
I hope you add the line to replace the image format to png if mask is None because the raw mask file format for voc is png.

`    if mask_form_file is not None:
        mask_form_file = mask_form_file.replace("images", "masks")
        mask = cv2.imread(mask_form_file, cv2.IMREAD_GRAYSCALE)
        if mask is None:
              mask_form_file = mask_form_file.replace("jpg", "png")
              mask = cv2.imread(mask_form_file, cv2.IMREAD_GRAYSCALE)
        mask = np.expand_dims(mask, axis=2)
 
Add a suggestion, <Ctrl+g>
Add heading textAdd bold text, <Ctrl+b>Add italic text, <Ctrl+i>
`

Reply:
According to Common Semantic Segmentation format - masks and images should be with same extension. So, it is not required. Other formats are not supported for this feature. Also, please, note that it is mainly for debugging, supporting some custom fine-grained and some comparison between mmseg and otx. So, I'm not sure that it is worth to spend time and support other formats.